lib/dfid-transition/patch/specialist_publisher/themes.rb

```diff
@@ -17,22 +17,35 @@ class Themes < Base
         SPARQL
 
         def mutate_schema
-          repository = RDF::Repository.load('http://r4d.dfid.gov.uk/RDF/SKOS/Themes.rdf')
-          sparql = SPARQL::Client.new(repository)
+          theme_facet['allowed_values'] = transform_to_label_value(
+            sorted_r4d_solutions
+          )
+        end
 
-          results = sparql.query(QUERY)
+      private
 
-          theme_facet['allowed_values'] = transform_to_label_value(results)
+        def sparql
+          SPARQL::Client.new(repository)
         end
 
-      private
+        def repository
+          RDF::Repository.load('http://r4d.dfid.gov.uk/RDF/SKOS/Themes.rdf')
+        end
+
+        def sorted_r4d_solutions
+          r4d_solutions.sort_by { |result| result['prefLabel'].value }
+        end
+
+        def r4d_solutions
+          sparql.query(QUERY)
+        end
```

Comment:
There are a few issues with 'methodising' everything like this:

* The return values of `sparql` and `repository` aren't memoized, so if they are referred to twice, they'd be re-instantiated. It's a reasonable assumption for someone modifying this class in future to make that if they refer to `repository` twice, the second time they refer to it it'll be the same instance. We'd need `@repository ||= ...` to memoize here.
* The method name `sparql` is a bit misleading – it isn't a bit of SPARQL, it's a `sparql_client`. This confusion is compounded by the fact we've got a `SPARQL` constant that *does* have SPARQL in it.
* Spreading these values out over the file makes things arguably less readable than the original one-shot method
* We pass `r4d_solutions` as a parameter to `transform_to_label_value` when we have a method of the same name in the class.
* We're querying the triples then sorting the solutions in Ruby; we could get the triplestore to do this for us by adding an `ORDER BY ?prefLabel` clause to the SPARQL.